src/meta/TableSpec.h

```diff
@@ -185,5 +202,58 @@ struct TableSpecEqual {
 
 using TableSpecSet = nebula::common::unordered_set<TableSpecPtr, TableSpecHash, TableSpecEqual>;
 
+// enum to string
+inline auto getVal = [](nebula::meta::PatternMacro p) {
+  switch (p) {
+  case nebula::meta::PatternMacro::DATE:
+    return "date";
+  case nebula::meta::PatternMacro::HOUR:
+    return "hour";
+  case nebula::meta::PatternMacro::MINUTE:
+    return "minute";
+  case nebula::meta::PatternMacro::SECOND:
+    return "second";
+  case nebula::meta::PatternMacro::TIMESTAMP:
+    return "timestamp";
+  default: return "";
+  }
+};
```

Comment:
would rather to have a static hash map to store this mapping. 
nebula:common::unordered_map is super fast - defined by https://github.com/varchar-io/nebula/blob/master/src/common/Hash.h#L169

Reply:
fixed.